src/test/java/org/mifos/integrationtest/cucumber/stepdef/GSMATransferStepDef.java

```diff
@@ -720,6 +720,49 @@ public void iCallTheBalanceApiForPayeeBalanceForCombinedTestsCases(String id) th
 
     }
 
+    @Then("I call the balance api for payer {string} balance for all combine test cases")
+    public void iCallTheBalanceApiForAllPayerBalanceForCombinedTestCases(String id) throws JsonProcessingException {
+        RequestSpecification requestSpec = Utils.getDefaultSpec(scenarioScopeState.tenant);
+        String finalEndpoint = amsBalanceEndpoint;
+        if (scenarioScopeState.payerIdentifierforBatch == null || id.equals("1")) {
+            scenarioScopeState.payerIdentifierforBatch = new String[15];
+        }
+        scenarioScopeState.payerIdentifierforBatch[Integer.parseInt(id)] = scenarioScopeState.payerIdentifier;
+        finalEndpoint = finalEndpoint.replace("{IdentifierType}", "MSISDN");
+        finalEndpoint = finalEndpoint.replace("{IdentifierId}",
+                (scenarioScopeState.debitParty == null) ? scenarioScopeState.payerIdentifier : scenarioScopeState.debitParty);        logger.info("Endpoint: " + finalEndpoint);
+        scenarioScopeState.response = RestAssured.given(requestSpec).baseUri(amsBaseUrl).body("").expect()
+                .spec(new ResponseSpecBuilder().expectStatusCode(200).build()).when().get(finalEndpoint).andReturn().asString();
+        logger.info("Balance Response: " + scenarioScopeState.response);
+        InteropAccountDTO interopAccountDTO = objectMapper.readValue(scenarioScopeState.response, InteropAccountDTO.class);
+        assertThat(
+                interopAccountDTO.getAvailableBalance().intValue() >= scenarioScopeState.initialBalForPayerForBatch[Integer.parseInt(id)])
+                .isTrue();
+
+    }
+
+
+    @Then("I call the balance api for payee {string} balance for all combine test cases")
+    public void iCallTheBalanceApiForPayeeBalanceForAllCombinedTestsCases(String id) throws JsonProcessingException {
+        RequestSpecification requestSpec = Utils.getDefaultSpec(scenarioScopeState.tenant);
```

Comment:
if some part of this code is a duplicate of the above, use a method to re-use the code